### PR TITLE
feat: Feature/cmp 7269 able to fetch children and parents to create hierarchys

### PR DESCRIPTION
**Pull request by dev-warner**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Tests (`npm run test`) for the changes have been added (for bug fixes / features)
- [x] Docs (`npm run doc`) have been reviewed and added / updated if needed (for bug fixes / features)
- [x] PR title and git commit messages conform to the [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0/) specification

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

current behaviour doesn't support, hierarchy requests.

## What is the new behaviour?

- able to fetch children by id
- able to fetch parents by id
- able to fetch children/parents from sub nodes
- able to fetch content link associated with node

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

- Changed readme badge to github actions.
- Changed eslint rule to support current use of any
- Fixed eslint warnings for unused vars
- Added some typings

### CODE REVIEW

File: src/lib/model/HierarchyChildren.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  public label: string;
+  public parentId: string | undefined;
+  public contentTypeUri: string;
+  public root: boolean;
+  public hasChildren: boolean;
+  public repositoryId: string;
+  public publishingStatus: ContentItemPublishingStatus | string;
+
+  /**
+   *
```
Comment (easen-amp) [inline src/lib/model/HierarchyChildren.ts:26]: Missing comment

File: src/lib/model/PublishingStatus.ts [hunk 2]
```diff
@@ -9,3 +9,9 @@ export enum PublishingStatus {
   UNSCHEDULING = 'UNSCHEDULING',
   PUBLISHING = 'PUBLISHING',
 }
+
+export enum ContentItemPublishingStatus {
```
Comment (easen-amp) [inline src/lib/model/PublishingStatus.ts:13]: I think this is missing a comment/doc

File: src/lib/model/HierarchyNode.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+   * Schema URI of the content item
+   */
+  contentTypeUri: string;
+
+  /**
+   * Boolean to denote weather the hierarchy node has children
+   */
+  hasChildren: boolean;
+
+  /**
```
Comment (easen-amp) [inline src/lib/model/HierarchyNode.ts:33]: Most of this comment should be on the `ContentItemPublishingStatus` enum IMHO

File: .eslintrc.yaml [hunk 4]
```diff
@@ -3,20 +3,21 @@ env:
...
-  - "@typescript-eslint/eslint-plugin"
-  - "unused-imports"
\ No newline at end of file
+  '@typescript-eslint/no-parameter-properties': 'off'
+  '@typescript-eslint/interface-name-prefix': 'off'
+  '@typescript-eslint/explicit-function-return-type': 'off'
+  '@typescript-eslint/no-explicit-any': 'off'
+  '@typescript-eslint/no-empty-function': 'off'
+  '@typescript-eslint/no-unused-vars': 'off'
+  '@typescript-eslint/explicit-module-boundary-types': 'off'
```
Comment (easen-amp) [inline .eslintrc.yaml:18]: I feel like we should have this turned on https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md

> Explicit types for function return values and arguments makes it clear to any calling code what is the module boundary's input and output.
>Consider using this rule in place of no-untyped-public-signature which has been deprecated.

I know this is the source of the eslint lint warnings (which should be addresses).

As this is an SDK, we should enforce that all exported methods have a defined return type, using actual types and not `any`.

**Review by easen-amp — CHANGES_REQUESTED** [review]

**Review by easen-amp — APPROVED** [review]

File: src/lib/DynamicContent.ts [hunk 5]
```diff
@@ -131,6 +133,34 @@ export class DynamicContent {
       ),
   };
 
+  /**
+   * Hierarchy Resources
+   */
+  public readonly hierarchys = {
```
Comment (nsmith-amplience) [inline src/lib/DynamicContent.ts:139]: typo, should be `hierarchies`

**Review by nsmith-amplience — APPROVED** [review]

**Review by simonmudd — APPROVED** [review]

**Review by Benj0s — APPROVED** [review]

**Pull request merged**